### PR TITLE
Feature: CorePKCS

### PR DESCRIPTION
**Pull request by lundinc2**
*Issue #, if available:*

*Description of changes:*
* Added corePKCS11 library as a submodule
* Added PKCS #11 system tests
* Added PKCS #11 demos

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/pkcs11/common/include/core_pkcs11_config.h [hunk 1]
```diff
@@ -0,0 +1,170 @@
+/*
+ * FreeRTOS V1.4.8
```
Comment (sarenameas) [inline demos/pkcs11/common/include/core_pkcs11_config.h:2]: We have our config files versioned with the CSDK

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+#endif
+
+#include "logging_stack.h"
+
+/* PKCS #11 includes. */
+#include "core_pkcs11.h"
+#include "pkcs11.h"
+
+/**
+ *@brief This function details how to use the PKCS #11 "Management" functions to
```
Comment (sarenameas) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:47]:
```suggestion
 * @brief This function details how to use the PKCS #11 "Management" functions to 
```

File: demos/pkcs11/common/include/core_pkcs11_config.h (hunk 1 above)
Comment (lundinc2) [reply demos/pkcs11/common/include/core_pkcs11_config.h]: Thanks

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 2 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c]: Thanks

File: .gitmodules [hunk 3]
```diff
@@ -19,3 +19,12 @@
 [submodule "libraries/aws/device-defender-for-aws-iot-embedded-sdk"]
 	path = libraries/aws/device-defender-for-aws-iot-embedded-sdk
 	url = https://github.com/aws/device-defender-for-aws-iot-embedded-sdk.git
+[submodule "libraries/standard/corePKCS11"]
+	path = libraries/standard/corePKCS11
+	url = https://github.com/FreeRTOS/corePKCS11.git
+[submodule "libraries/3rdparty/mbedtls"]
+	path = libraries/3rdparty/mbedtls
+	url = https://github.com/ARMmbed/mbedtls.git
```
Comment (lundinc2) [inline .gitmodules:27]: Need to update the CMake to add this as a library

File: libraries/CMakeLists.txt [hunk 4]
```diff
@@ -12,9 +12,25 @@ macro( clone_path path )
     endif()
 endmacro()
 
+# Macro utility to checkout a submodule that has the update flag set to none
```
Comment (aggarw13) [inline libraries/CMakeLists.txt:15]: Possible to update the existing `clone_path` macro by taking an extra parameter to indicate whether `--checkout` flag should be used? 
If the caller does not provide the extra parameter, then the default behavior of cloning can be without the `--checkout` flag

File: integration-test/pkcs11/test-config/core_pkcs11_config.h [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+/* Logging configuration for the PKCS #11 library. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "PKCS11"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_NONE /* PKCS #11 MBT tests will generate a lot of logs and causes parsing errors, that is why this is defaulted to NONE. */
+#endif
+
+#include "logging_stack.h"
```
Comment (aggarw13) [inline integration-test/pkcs11/test-config/core_pkcs11_config.h:57]: Can you add the footer comment about ending logging configuration to all the config files?
```suggestion
#include "logging_stack.h"

/************ End of logging configuration ****************/

```

File: integration-test/pkcs11/test-config/core_pkcs11_config.h [hunk 6]
```diff
@@ -0,0 +1,176 @@
+/*
+ * FreeRTOS V1.1.4
```
Comment (aggarw13) [inline integration-test/pkcs11/test-config/core_pkcs11_config.h:2]: Please make the change in all files
```suggestion
 * AWS IoT Device SDK for Embedded C
```

File: integration-test/pkcs11/test-config/core_pkcs11_config.h [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+ */
+
+/**
+ * @file aws_pkcs11_config.h
+ * @brief PCKS#11 config options.
+ */
+
+
+#ifndef _AWS_PKCS11_CONFIG_H_
+#define _AWS_PKCS11_CONFIG_H_
```
Comment (aggarw13) [inline integration-test/pkcs11/test-config/core_pkcs11_config.h:33]:
```suggestion
#ifndef CORE_PKCS11_CONFIG_H_
#define CORE_PKCS11_CONFIG_H_
```

File: integration-test/pkcs11/test-config/core_pkcs11_config.h [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file aws_pkcs11_config.h
```
Comment (aggarw13) [inline integration-test/pkcs11/test-config/core_pkcs11_config.h:27]:
```suggestion
 * @file core_pkcs11_config.h
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    PROPERTIES COMPILE_FLAGS
+    "-Wno-pedantic"
+)
+
+list(APPEND PKCS_SOURCES
+            "${MBEDTLS_FILES}"
+            "${COREPKCS11_LOCATION}/source/portable/posix/core_pkcs11_pal.c"
+            "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_utils.c"
+            "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_error.c"
+            )
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt:22]: Indentation fix
```suggestion
list(APPEND PKCS_SOURCES
    "${MBEDTLS_FILES}"
    "${COREPKCS11_LOCATION}/source/portable/posix/core_pkcs11_pal.c"
     "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_utils.c"
     "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_error.c"
)
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    ${DEMO_NAME}
+    PUBLIC
+        -DMBEDTLS_CONFIG_FILE="mbedtls_config.h"
+)
+
+
+target_include_directories(
+    ${DEMO_NAME}
+    PUBLIC
+        "../common/include"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt:41]:
```suggestion
        "${DEMOS_DIR}/pcks11/common/include"
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        ${PKCS_INCLUDE_PUBLIC_DIRS}
+        ${CMAKE_CURRENT_LIST_DIR}
+        ${LOGGING_INCLUDE_DIRS}
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/pkcs11"
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/include"
+    PRIVATE
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/CMakeLists.txt:48]: (Nitpick)
```suggestion
        ${CORE_PKCS11_3RDPARTY_LOCATION}/pkcs11
        ${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/include
    PRIVATE
        ${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+/**
+ * @brief Entry point of demo.
+ *
+ * The example shown above uses PKCS #11 APIs to manage state and generate random numbers.
+ */
+int main( void )
+{
+    vPKCS11ManagementAndRNGDemo();
+    return 0;
+}
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:238]: Any reason for not having the implementation of `vPKCS11ManagementAndRNGDemo` function within `main` itself?

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+/* PKCS #11 includes. */
+#include "core_pkcs11.h"
+#include "pkcs11.h"
+
+/**
+ * @brief This function details how to use the PKCS #11 "Management" functions to
+ * manage the internal state machine of the PKCS #11 implementation. These
+ * functions are all defined in
+ * http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html
+ * please consult the standard for more information regarding these functions.
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:51]:
```suggestion
 * http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html
 * Please consult the standard for more information regarding these functions.
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+    assert( pxFunctionList != NULL );
+    assert( pxFunctionList->C_Initialize != NULL );
+    assert( pxFunctionList->C_GetSlotList != NULL );
+    assert( pxFunctionList->C_OpenSession != NULL );
+    assert( pxFunctionList->C_Login != NULL );
+    assert( pxFunctionList->C_GenerateRandom != NULL );
+    assert( pxFunctionList->C_CloseSession != NULL );
+    assert( pxFunctionList->C_Finalize != NULL );
+
+    LogInfo( ( "Cryptoki Major Version: %u Minor Version %u",
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:122]:
```suggestion
    LogInfo( ( "Cryptoki Major Version: %u Minor Version: %u",
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+    /* We will use the terminology as defined in the standard, Cryptoki is in
+     * reference to the Cryptographic Token Interface defined in the PKCS #11
+     * standard. An implementation of Cryptoki is referred to as a
+     * "Cryptoki library". */
+    LogInfo( ( "Starting PKCS #11 Management and Random Number Generation" \
+               " Demo." ) );
+
+    /* CK_RV is the return type for a Cryptoki function. Generally the underlying
+     * type is a CK_ULONG, it can also be a CKR_VENDOR_DEFINED type. */
+    CK_RV xResult = CKR_OK;
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:70]: Hungarian notation of variable naming is not used in C-SDK. (Demo and test wide changes can be made in a separate PR though)

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+    assert( pxFunctionList->C_Finalize != NULL );
+
+    LogInfo( ( "Cryptoki Major Version: %u Minor Version %u",
+               pxFunctionList->version.major,
+               pxFunctionList->version.minor ) );
+
+    /* C_Initialize will initialize the Cryptoki library and the hardware it
+     * abstracts. */
+    xResult = pxFunctionList->C_Initialize( &xInitArgs );
+    assert( xResult == CKR_OK );
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:129]: C-SDK demos do not use asserts but return codes to indicate failure. Is it possible for you to change the style for the demo to be consistent with other demos?

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+
+/**
+ * @brief This function details how to use the PKCS #11 "Management" functions to
+ * manage the internal state machine of the PKCS #11 implementation. These
+ * functions are all defined in
+ * http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html
+ * please consult the standard for more information regarding these functions.
+ *
+ * The standard has grouped the functions presented in this demo as:
+ * General Purpose Functions
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:54]: Can you list the specific functions used in the demo for each group?
It may provide better value to the reader..

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+     * NULL should always be passed as the argument, as the parameter is currently
+     * just reserved for future revisions.
+     *
+     * Calling this function in a multi threaded environment can lead to undefined
+     * behavior if other threads are accessing the Cryptoki library. */
+    xResult = pxFunctionList->C_Finalize( NULL );
+    assert( xResult == CKR_OK );
+
+    LogInfo( ( "Finished PKCS #11 Management and Random Number Generation" \
+               " Demo." ) );
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:221]: Escapes are not required for macro calls!
```suggestion
    LogInfo( ( "Finished PKCS #11 Management and Random Number Generation" 
               " Demo." ) );
```

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+ */
+
+/* Standard includes. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+
+/* Logging configuration for the PKCS #11 library. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "PKCS11_DEMO"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:33]: Would be better to have differentiating log names for the 2 demos

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/pkcs11_demo_mechanisms_and_digests.c [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+        pcFormatPtr += sprintf( ( char * ) pcFormatPtr, "%x", xDigestResult[ ulIndex ] );
+    }
+
+    LogInfo( ( "Hash of known message using SHA256: %s.", xHashFormatBuffer ) );
+    LogInfo( ( "Finished PKCS #11 Mechanisms and Digest Demo." ) );
+
+    /* Avoid compiler warnings if asserts are disabled. */
+    ( void ) xResult;
+
+    vEnd( hSession, pxSlotId );
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_mechanisms_and_digests/pkcs11_demo_mechanisms_and_digests.c:271]: Does one of the PKCS libraries expose this function?
If relevant in the demo, a comment about it would be useful

File: demos/pkcs11/pkcs11_demo_sign_and_verify/CMakeLists.txt [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+file(GLOB MBEDTLS_FILES CONFIGURE_DEPENDS "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/library/*.c")
+set_source_files_properties(
+    ${MBEDTLS_FILES}
+    PROPERTIES COMPILE_FLAGS
+    "-Wno-pedantic"
+)
+
+list(APPEND PKCS_SOURCES
+            "${MBEDTLS_FILES}"
+            "../common/src/demo_helpers.c"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_sign_and_verify/CMakeLists.txt:20]:
```suggestion
            "${DEMOS_DIR}/pkcs11/common/src/demo_helpers.c"
```

File: demos/pkcs11/pkcs11_demo_sign_and_verify/CMakeLists.txt [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+target_compile_definitions(
+    ${DEMO_NAME}
+    PUBLIC
+        -DMBEDTLS_CONFIG_FILE="mbedtls_config.h"
+)
+
+target_include_directories(
+    ${DEMO_NAME}
+    PUBLIC
+        "../common/include"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_sign_and_verify/CMakeLists.txt:42]:
```suggestion
        "${DEMOS_DIR}/pkcs11/common/include"
```

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+file(GLOB MBEDTLS_FILES CONFIGURE_DEPENDS "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/library/*.c")
+set_source_files_properties(
+    ${MBEDTLS_FILES}
+    PROPERTIES COMPILE_FLAGS
+    "-Wno-pedantic"
+)
+
+list(APPEND PKCS_SOURCES
+            "${MBEDTLS_FILES}"
+            "../common/src/demo_helpers.c"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt:20]:
```suggestion
            "${DEMOS_DIR}/pkcs11/common/src/demo_helpers.c"
```

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+    ${DEMO_NAME}
+    PUBLIC
+        -DMBEDTLS_CONFIG_FILE="mbedtls_config.h"
+)
+
+
+target_include_directories(
+    ${DEMO_NAME}
+    PUBLIC
+        "../common/include"
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt:43]:
```suggestion
        "${DEMOS_DIR}/pkcs11/common/include"
```

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt [hunk 25]
```diff
@@ -0,0 +1,52 @@
+set( DEMO_NAME "pkcs11_demo_mechanisms_and_digests" )
+
+# Set path to corePKCS11 and it's third party libraries.
+set(COREPKCS11_LOCATION "${CMAKE_SOURCE_DIR}/libraries/standard/corePKCS11")
+set(3RDPARTY_LOCATION "${CMAKE_SOURCE_DIR}/libraries/3rdparty")
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt:5]: This seems to be unused in the file
```suggestion
set(3RDPARTY_LOCATION "${CMAKE_SOURCE_DIR}/libraries/3rdparty")
```

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt [hunk 26]
```diff
@@ -0,0 +1,52 @@
+set( DEMO_NAME "pkcs11_demo_mechanisms_and_digests" )
+
+# Set path to corePKCS11 and it's third party libraries.
```
Comment (aggarw13) [inline demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt:3]: (Nit)
```suggestion
# Set path to corePKCS11 and its third party libraries.
```

File: demos/pkcs11/common/src/demo_helpers.c [hunk 27]
```diff
@@ -0,0 +1,270 @@
...
+/**
+ * @file demo_helpers.c
+ * @brief Common functions used between the PKCS #11 demos.
+ */
+/* Standard includes. */
+#include <stdio.h>
+#include <assert.h>
+
+/* PKCS #11 includes. */
+#include "core_pkcs11_config.h"
```
Comment (aggarw13) [inline demos/pkcs11/common/src/demo_helpers.c:35]: If the corePKCS11 library is the main consumer of the config file, does this file need to include it?

File: libraries/CMakeLists.txt (hunk 4 above)
Comment (lundinc2) [reply libraries/CMakeLists.txt:15]: Will add this.

File: integration-test/pkcs11/test-config/core_pkcs11_config.h (hunk 5 above)
Comment (lundinc2) [reply integration-test/pkcs11/test-config/core_pkcs11_config.h:57]: Will add this

File: integration-test/pkcs11/test-config/core_pkcs11_config.h (hunk 6 above)
Comment (lundinc2) [reply integration-test/pkcs11/test-config/core_pkcs11_config.h:2]: Ah right, thank you

File: demos/pkcs11/common/src/demo_helpers.c (hunk 27 above)
Comment (lundinc2) [reply demos/pkcs11/common/src/demo_helpers.c:35]: It is being used for `configPKCS11_DEFAULT_USER_PIN` but I think I will remove the dependency, and hard code the login PIN, as our implementation doesn't support it.

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 19 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:33]: Okay I will make them demo specific

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 17 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:54]: Sure I will add that

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 15 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:70]: Hmm okay, that will be a big change so agreed, separate PR will be best

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 16 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:129]: Yes I can update to do this

File: demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c (hunk 12 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c:238]: In FreeRTOS/FreeRTOS the demos have the same start point, and are not separated out like this

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt (hunk 25 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_mechanisms_and_digests/CMakeLists.txt:5]: Oops, was leftover from older work

File: demos/pkcs11/pkcs11_demo_mechanisms_and_digests/pkcs11_demo_mechanisms_and_digests.c (hunk 20 above)
Comment (lundinc2) [reply demos/pkcs11/pkcs11_demo_mechanisms_and_digests/pkcs11_demo_mechanisms_and_digests.c:271]: This is for convenience to not repeat code explained in the first demo

File: demos/lexicon.txt [hunk 28]
```diff
@@ -1,109 +1,271 @@
...
+ccs
 cdn
+cer
 cert
+certificatetemplate
 certs
+cfb
+chacha
+chachapoly
+chinese
```
Comment (sarenameas) [inline demos/lexicon.txt:65]: Woah is "chinese" not a dictionary word? lol

File: libraries/CMakeLists.txt (hunk 4 above)
Comment (lundinc2) [reply libraries/CMakeLists.txt]: Added optional parameter to command string

File: demos/lexicon.txt (hunk 28 above)
Comment (lundinc2) [reply demos/lexicon.txt:65]: Haha I did not notice that

**Review by aggarw13 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]
